scikeras/wrappers.py

```diff
@@ -1310,6 +1316,24 @@ def _type_of_target(self, y: np.ndarray) -> str:
             target_type = type_of_target(self.classes_)
         return target_type
 
+    def _fit_keras_model(self, *args, **kwargs):
+        try:
+            super()._fit_keras_model(*args, **kwargs)
+        except ValueError as e:
+            if (
+                self.loss == "categorical_crossentropy"
+                and hasattr(self, "model_")
+                and 1 in {o.shape[1] for o in getattr(self.model_, "outputs", [])}
+            ):
+                raise ValueError(
+                    "The model is configured to have one output, but the "
+                    f"loss='{self.loss}' is expecting multiple outputs "
+                    "(which is often used with one-hot encoded targets). "
+                    "More detail on Keras losses: https://keras.io/api/losses/"
+                ) from e
+            else:
+                raise e
```

Comment:
This error message only provides marginal utility: it protects against cases when the model has one output but there are multiple classes.

It can not go in `_check_model_compatibility`; I wait for an error to be raised before issuing this warning (otherwise a model with a single output raises an error).

Reply:
Got it. Is there a specific error message we can check for, like `if "some Keras error" in str(e)`?

> `getattr(self.model_, "outputs", [])`

Is this necessary? `model_` should always have an `outputs` attribute, except in the case described in #207, but that should be a separate check/error.

> f"loss='{self.loss}' is expecting multiple outputs "

Can you clarify what you mean by a loss expecting a number of outputs? My understanding is that Keras "broadcasts" losses to outputs, so if you give it a scalar loss (ie.. `loss="bce"`) with 2 outputs (i.e. `len(model_.outputs) == 2`), it will implicitly compile the model with `loss=[original_loss] * len(outputs)`. But you can actually map losses to outputs manually, by passing `loss=["bce", "mse"]` or `loss={"out1": "bce", "out2": "mse"}`. From the tests, it seems like by "loss is expecting multiple outputs" you mean that there is a single output unit but multiple classes, which I feel like could be confused with the above concept of configuring a loss for multiple outputs.

I'm also curious about the iteration through outputs (`1 in {o.shape[1] for o in self.model_.outputs}`). SciKeras does not support >1 output out of the box (users need to override `target_encoder`) so it seems a bit strange to try to account for that when using the default loss. I feel that using the default loss should only be supported for the simple single-output cases that `target_encoder` supports.

Reply:
As a side note: I think giving users better errors and validating their inputs like you are doing here can be a very valuable part of SciKeras, but currently it is done in an ad-hoc manner via `_check_model_compatibility`, etc. I think if we add more of these types of things, it would be nice to have an organized interface for it. I opened #209 to try to brainstorm ideas for this.